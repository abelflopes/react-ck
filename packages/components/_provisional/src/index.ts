export * from "./image";

export * from "./avatar";

export * from "./dropdown";

export * from "./dock";

export * from "./breadcrumbs";

export * from "./scrollable-container";

export * from "./position-engine";

export * from "./top-bar";

export * from "./snackbar";

export * from "./overlay-spinner";

export * from "./menu";

export * from "./tooltip";

export * from "./popover";

export * from "./chat";

export * from "./file-uploader";

export * from "./attachment";

export * from "./snippet";
