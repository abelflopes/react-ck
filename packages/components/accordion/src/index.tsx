import React from "react";
import { List, type ListProps } from "@react-ck/list";
import { Collapse } from "@react-ck/collapse";
import styles from "./styles/index.module.scss";

export interface AccordionItem {
  header: NonNullable<React.ReactNode>;
  children: NonNullable<React.ReactNode>;
}

export interface AccordionProps extends Omit<ListProps, "items"> {
  items: AccordionItem[];
}

/**
 * An accordion is a vertically stacked list of interactive items. Each item can be "collapsed" with only a summary visible or “expanded” to show the full content for that item.
 * @param props - {@link AccordionProps}
 * @returns a React element
 */

export const Accordion = ({
  items,
  ...otherProps
}: Readonly<AccordionProps>): React.ReactElement => (
  <List
    {...otherProps}
    items={items.map(({ header, children }) => (
      <Collapse key={String(JSON.stringify(header))} header={header}>
        <div className={styles.content}>{children}</div>
      </Collapse>
    ))}
  />
);
