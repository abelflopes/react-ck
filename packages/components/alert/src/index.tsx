import React from "react";
import classNames from "classnames";
import styles from "./styles/index.module.scss";
import { Text } from "@react-ck/text";

export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
  heading?: string;
  /** Specifies the visual style of the alert  */
  skin?: "neutral" | "primary" | "negative" | "average" | "positive";
}

// TODO: add dismiss

/**
 * Alert is a short message that provides contextual feedback in a prominent way.
 * @param props - {@link AlertProps}
 * @returns a React element
 */

export const Alert = ({
  heading,
  skin = "neutral",
  children,
  className,
  ...otherProps
}: Readonly<AlertProps>): React.ReactElement => (
  <div {...otherProps} className={classNames(styles.root, styles[skin], className)}>
    {heading && (
      <Text type="h4" as="p" variation="bold" margin={false} className={styles.heading}>
        {heading}
      </Text>
    )}
    {children}
  </div>
);
