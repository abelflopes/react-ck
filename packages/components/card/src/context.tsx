import React from "react";

export interface CardContextValue {
  image: React.ImgHTMLAttributes<HTMLImageElement> | undefined;
}

export interface CardContextProps {
  setValue: (value: CardContextValue) => void;
}

export const themeContextDefaults: CardContextProps = {
  setValue: () => undefined,
};

export const CardContext = React.createContext<CardContextProps>(themeContextDefaults);
