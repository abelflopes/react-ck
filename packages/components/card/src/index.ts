export * from "./Card";

export * from "./CardImage";
