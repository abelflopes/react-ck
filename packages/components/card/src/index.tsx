import classNames from "classnames";
import styles from "./styles/index.module.scss";
import React from "react";

export const Card = ({
  children,
  className,
  ...otherProps
}: Readonly<React.HTMLAttributes<HTMLDivElement>>): React.ReactElement => (
  <div {...otherProps} className={classNames(styles.root, className)}>
    {children}
  </div>
);
