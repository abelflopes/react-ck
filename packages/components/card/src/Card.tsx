import classNames from "classnames";
import styles from "./styles/index.module.scss";
import React, { useMemo, useState } from "react";
import { CardContext, type CardContextValue } from "./context";

export const Card = ({
  children,
  className,
  ...otherProps
}: Readonly<React.HTMLAttributes<HTMLDivElement>>): React.ReactElement => {
  const [contextValue, setContextValue] = useState<CardContextValue>({
    image: undefined,
  });

  const contextProps = useMemo(
    () => ({
      setValue: setContextValue,
    }),
    [],
  );

  return (
    <CardContext.Provider value={contextProps}>
      <div {...otherProps} className={classNames(styles.root, className)}>
        {contextValue.image && (
          <img
            {...contextValue.image}
            alt={contextValue.image.alt}
            className={classNames(styles.image, contextValue.image.className)}
          />
        )}
        <div className={styles.content}>{children}</div>
      </div>
    </CardContext.Provider>
  );
};
