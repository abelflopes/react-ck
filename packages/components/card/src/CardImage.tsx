import { useContext, useEffect } from "react";
import { CardContext, type CardContextValue } from "./context";

export const CardImage = (props: NonNullable<CardContextValue["image"]>): undefined => {
  const context = useContext(CardContext);

  useEffect(() => {
    context.setValue({ image: props });

    return () => {
      context.setValue({ image: undefined });
    };
  }, [props]);
};
