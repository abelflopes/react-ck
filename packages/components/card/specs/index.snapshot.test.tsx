import React from "react";
import { Card } from "../src/index";
import renderer from "react-test-renderer";

describe("Snapshot Card", () => {
  test("renders correctly", async () => {
    const tree = renderer.create(<Card>Card</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
