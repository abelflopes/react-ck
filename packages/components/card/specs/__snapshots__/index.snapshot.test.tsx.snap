// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Card renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="content"
  >
    Card
  </div>
</div>
`;
